test/integration/provider/azure.go

```diff
@@ -186,12 +155,31 @@ func getOrphanedVMs(
 
 	for _, virtualMachine := range virtualMachines.Values() {
 		if value, ok := virtualMachine.Tags[ITResourceTagKey]; ok && *value == ITResourceTagValue {
-			err := deleteVM(clients, resourceGroup, *virtualMachine.Name)
-			if err != nil {
-				orphanedVMs = append(orphanedVMs, *virtualMachine.Name)
-			}
+			orphanedVMs = append(orphanedVMs, *virtualMachine.Name)
 		}
 	}
 
 	return orphanedVMs, nil
 }
+
+func cleanUpOrphanedResources(orphanedVms []string, orphanedVolumes []string, orphanedNICs []string, clients spi.AzureDriverClientsInterface, resourceGroup string) (delErrOrphanedVms []string, delErrOrphanedVolumes []string, delErrOrphanedNICs []string) {
+	for _, virtualMachineName := range orphanedVms {
+		err := deleteVM(clients, resourceGroup, virtualMachineName)
+		if err != nil {
+			delErrOrphanedVms = append(delErrOrphanedVms, virtualMachineName)
+		}
+	}
+	for _, volumeId := range orphanedVolumes {
+		err := deleteDisk(clients, resourceGroup, volumeId)
+		if err != nil {
+			delErrOrphanedVolumes = append(delErrOrphanedVolumes, volumeId)
+		}
+	}
+	for _, networkInterfaceName := range orphanedNICs {
+		err := deleteNICs(clients, resourceGroup, networkInterfaceName)
+		if err != nil {
+			delErrOrphanedNICs = append(delErrOrphanedNICs, networkInterfaceName)
+		}
+	}
+	return
```

Comment:
why are we not returning the lists of orphan resources which cannot be deleted

Reply:
named return parameters are returned like this only.